cloudflare/data_source_zone.go

```diff
@@ -0,0 +1,109 @@
+package cloudflare
+
+import (
+	"context"
+	"fmt"
+	"github.com/cloudflare/cloudflare-go"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"log"
+)
+
+func dataSourceCloudflareZone() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceCloudflareZoneRead,
+
+		Schema: map[string]*schema.Schema{
+			"zone_id": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+			},
+			"account_id": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+			},
+			"name": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+			},
```

Comment:
looking at the use cases, i don't think it makes sense to ever apply these two together. if you want clarity, zone ID is unique and should be enough. what was your use case here?

```suggestion
			"zone_id": {
				Type:     schema.TypeString,
				Optional: true,
				Computed: true,
				ConflictsWith: []string{"name"},
			},
			"account_id": {
				Type:     schema.TypeString,
				Optional: true,
				Computed: true,
			},
			"name": {
				Type:     schema.TypeString,
				Optional: true,
				Computed: true,
				ConflictsWith: []string{"zone_id"},
			},
```

Reply:
Oh sweet! I didn't know there was a `ConflictsWith` option

Reply:
np! check out everything that is available at https://github.com/hashicorp/terraform-plugin-sdk/blob/main/helper/schema/schema.go#L196 if you're curious what else a schema can hold